### PR TITLE
moved error handling to c_utilities

### PR DESCRIPTION
**Pull request by wjwwood**
Connects to ros2/c_utilities#16

### CODE REVIEW

File: rmw/include/rmw/error_handling.h [hunk 1]
```diff
@@ -23,95 +23,38 @@ extern "C"
 #include <stdbool.h>
 #include <stddef.h>
 
-#include "rmw/macros.h"
```
Comment (mikaelarguedas) [inline rmw/include/rmw/error_handling.h]: Should the macros (STRINGIFY and friends) be typedefed to their c_utility equivalent ?

Comment (mikaelarguedas) [inline rmw/include/rmw/error_handling.h]: maybe this is captured in https://github.com/ros2/c_utilities/pull/16#issuecomment-294263596, if so feel free to discard this comment

Comment (wjwwood) [inline rmw/include/rmw/error_handling.h]: No, I missed that, I also missed removing the thread local stuff, which isn't used in RMW anymore. Good catch, see: 93d9825

File: rmw/include/rmw/error_handling.h [hunk 2]
```diff
@@ -23,95 +23,38 @@ extern "C"
...
 #include "rmw/visibility_control.h"
+#include "c_utilities/error_handling.h"
```
Comment (wjwwood) [inline rmw/include/rmw/error_handling.h]: fc2d4ea

**Pull request merged**